x-pack/legacy/plugins/ml/public/jobs/new_job_new/pages/components/pick_fields_step/components/single_metric_view/settings.tsx

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { Fragment, FC, useContext, useEffect, useState } from 'react';
+import { JobCreatorContext } from '../../../job_creator_context';
+import { BucketSpan } from '../bucket_span';
+
+interface Props {
+  isActive: boolean;
+  setIsValid: (proceed: boolean) => void;
+}
+
+export const SingleMetricSettings: FC<Props> = ({ isActive, setIsValid }) => {
+  const { jobCreator, jobCreatorUpdate, jobCreatorUpdated } = useContext(JobCreatorContext);
+  const [bucketSpan, setBucketSpan] = useState(jobCreator.bucketSpan);
+
+  useEffect(() => {
```

Comment:
not for this PR but just a general idea seeing this combination of `useState/useEffect` - maybe `jobCreator` could become a custom hook `useJobCreator`. Then you could do `const { bucketSpan, setBucketSpan } = useJobCreator();` and what's in the two following `useEffect` could become an implementation detail of `useJobCreator()`.

Reply:
I looked into creating a custom hook for this, i gave up as it didn't really add much value and i felt it hid away some of the functionality.
I'll revisit it to see if makes an improvement.